src/libmongoc/src/mongoc/mongoc-cluster-aws.c

```diff
@@ -350,6 +405,17 @@ _obtain_creds_from_ecs (_mongoc_aws_credentials_t *creds, bson_error_t *error)
       ecs_session_token = bson_iter_utf8 (&iter, NULL);
    }
 
+   mcd_timer expiration_timer;
+   bool expiration_set;
```

Comment:
These can be dropped and the values written directly into `creds->expiration`, IIUC.